dbms/src/Storages/DeltaMerge/Filter/In.h

```diff
@@ -59,6 +59,19 @@ class In : public RSOperator
         return rs_index ? rs_index->minmax->checkIn(start_pack, pack_count, values, rs_index->type)
                         : RSResults(pack_count, RSResult::Some);
     }
+
+    ColumnRangePtr buildSets(const LocalIndexInfosSnapshot & index_info) override
+    {
+        if (auto set = IntegerSet::createValueSet(attr.type->getTypeId(), values); set)
+        {
+            auto iter = std::find_if(index_info->begin(), index_info->end(), [&](const auto & info) {
+                return info.column_id == attr.col_id && info.kind == TiDB::ColumnarIndexKind::Inverted;
+            });
+            if (iter != index_info->end())
+                return SingleColumnRange::create(iter->column_id, iter->index_id, set);
+        }
```

Comment:
If the user write thousands or more values in the value list of `in (...)`. Would it bring OOM issue or performance regression? I think this can be an issue that you should verify later.

Reply:
For OOM, in this case, the bottleneck would not be the `ColumnRange`.

For performance, we will decide whether to use inverted index in optimizer based on the selectivity.